src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -161,16 +166,24 @@ g.test('stale')
 
     let device: GPUDevice | undefined = undefined;
     const promise = t.requestDeviceTracked(adapter);
-    if (awaitSuccess) {
-      device = await promise;
-      assert(device !== null);
+    if (awaitRequest2) {
+      if (request1 === 'Valid') {
+        await assertReject('OperationError', promise);
+      } else {
+        device = await promise;
+        assert(device !== null);
+      }
     } else {
-      t.shouldResolve(
-        (async () => {
-          const device = await promise;
-          device.destroy();
-        })()
-      );
+      if (request1 === 'Valid') {
+        t.shouldReject('OperationError', promise);
+      } else {
+        t.shouldResolve(
+          (async () => {
+            const device = await promise;
+            device.destroy();
+          })()
+        );
+      }
     }
 
     const kTimeoutMS = 1000;
```

Comment:
My understanding is that the the following promise `const lostDevice = await t.requestDeviceTracked(adapter);` will always fail now because we already have called `const promise = t.requestDeviceTracked(adapter);` earlier and  only one device can be requested from an adapter according to the spec. Shall we remove this bit?

Reply:
Wait, so sorry, I'm very confused. I didn't realize we already had a check down here that tests that the adapter is consumed. If we're already testing this, what test coverage are you trying to add with this CL?

Reply:
According to https://gpuweb.github.io/gpuweb/#ref-for-dom-adapter-state-consumed%E2%91%A1, the adapter must not be "consumed" when requesting a device, otherwise it should return a rejected promise with OperationError, not resolving with a lost device. The following assumes we'll always get a lost device. This is not true according to the spec.

FYI I'm working on https://dawn-review.googlesource.com/c/dawn/+/237274 to fix this behavior in Chromium.

Reply:
Ah right, that's because we changed the spec. But isn't this bit at the end of the test trying to specifically test the "consume"/"expire" behavior? Why not just update it so it expects a rejection instead of a device loss? (Won't this be necessary anyway, to make the test passable?)

Reply:
I've changed it to reject instead of device loss as advised in https://github.com/gpuweb/cts/pull/4336#discussion_r2053195855.
It works locally with https://dawn-review.googlesource.com/c/dawn/+/237274

<img width="1840" alt="image" src="https://github.com/user-attachments/assets/939fc70c-d6ab-4304-9c6f-11992cb17be9" />